data-prepper-plugins/translate-processor/src/main/java/org/opensearch/dataprepper/plugins/processor/translate/TargetsParameterConfig.java

```diff
@@ -0,0 +1,204 @@
+    @JsonProperty("target_type")
+    private TargetType targetType = TargetType.STRING;
+
+    public TargetsParameterConfig(Map<String, Object> map, String target, RegexParameterConfiguration regexParameterConfig, String translateWhen, String defaultValue, TargetType targetType) {
+        this.targetType = Optional.ofNullable(targetType).orElse(TargetType.STRING);
+        this.target = target;
+        this.map = map;
+        this.defaultValue = defaultValue;
+        this.regexParameterConfig = regexParameterConfig;
+        this.converter = this.targetType.getTargetConverter();
+        this.translateWhen = translateWhen;
+        parseMappings();
+    }
+
+    public void parseMappings() {
+        if (Objects.nonNull(getRegexParameterConfiguration())) {
+            compilePatterns(getRegexParameterConfiguration().getPatterns());
+        }
+        processMapField(map);
+        checkOverlappingKeys();
+    }
+
+    public String getTarget() {
+        return target;
+    }
+
+    public String getDefaultValue() {
+        return defaultValue;
+    }
+
+    public String getTranslateWhen() {
+        return translateWhen;
+    }
+
+    public TargetType getTargetType() {
+        return targetType;
+    }
+
+    public RegexParameterConfiguration getRegexParameterConfiguration() {
+        return regexParameterConfig;
+    }
+
+    public Map<String, Object> fetchIndividualMappings() {
+        return individualMappings;
+    }
+
+    public LinkedHashMap<Range<Float>, Object> fetchRangeMappings() {
+        return rangeMappings;
+    }
+
+    public Map<Pattern, Object> fetchCompiledPatterns() {
+        return compiledPatterns;
+    }
+
+    public TypeConverter getConverter() {
+        return converter;
+    }
+
+
+    @AssertTrue(message = "pattern option is mandatory while configuring regex option")
+    public boolean isPatternPresent() {
+        return regexParameterConfig == null || regexParameterConfig.getPatterns() != null;
+    }
+
+    @AssertTrue(message = "Either map or patterns option needs to be configured under targets.")
+    public boolean hasMappings() {
+        return Stream.of(map, regexParameterConfig).filter(n -> n != null).count() != 0;
+    }
+
+    @AssertTrue(message = "The mapped values do not match the target type provided")
+    public boolean isMapTypeValid() {
+        return map.keySet().stream().allMatch(key -> checkTargetValueType(map.get(key)));
+    }
+
+    @AssertTrue(message = "The pattern values do not match the target type provided")
+    public boolean isPatternTypeValid() {
+        if (Objects.isNull(regexParameterConfig) || Objects.isNull(regexParameterConfig.getPatterns())) {
+            return true;
+        }
+        Map<String, Object> patterns = regexParameterConfig.getPatterns();
+        return patterns.keySet().stream().allMatch(key -> checkTargetValueType(patterns.get(key)));
+    }
+
+    private boolean checkTargetValueType(Object val) throws NumberFormatException {
```

Comment:
This is a config file. Looks like a lot of these functions do not belong here. Convention is to put only config related functions in the `XXXConfig.java` file

Reply:
I will move the helper functions to a separate class